palace/fem/interpolator.cpp

```diff
@@ -107,4 +120,184 @@ std::vector<std::complex<double>> InterpolationOperator::ProbeField(const GridFu
   }
 }
 
+namespace fem
+{
+
+void InterpolateFunction(const mfem::GridFunction &U, mfem::GridFunction &V)
+{
+#if defined(MFEM_USE_GSLIB)
+  // Generate list of points where the grid function will be evaluated. If the grid function
+  // to interpolate is an H1 space of the same order as the mesh nodes, we can use the
+  // mesh node points directly. Otherwise, for a different basis order or type, we generate
+  // the interpolation points in the physical space manually.
+  auto &dest_mesh = *V.FESpace()->GetMesh();
+  MFEM_VERIFY(dest_mesh.GetNodes(), "Destination mesh has no nodal FE space!");
+  const int dim = dest_mesh.SpaceDimension();
+  mfem::Vector xyz;
+  mfem::Ordering::Type ordering;
+  const auto *dest_fec_h1 =
+      dynamic_cast<const mfem::H1_FECollection *>(V.FESpace()->FEColl());
+  const auto *dest_nodes_h1 = dynamic_cast<const mfem::H1_FECollection *>(
+      dest_mesh.GetNodes()->FESpace()->FEColl());
+  int dest_fespace_order = V.FESpace()->GetMaxElementOrder();
+  int dest_nodes_order = dest_mesh.GetNodes()->FESpace()->GetMaxElementOrder();
+  dest_mesh.GetNodes()->HostRead();
+  if (dest_fec_h1 && dest_nodes_h1 && dest_fespace_order == dest_nodes_order)
+  {
+    xyz.MakeRef(*dest_mesh.GetNodes(), 0, dest_mesh.GetNodes()->Size());
+    ordering = dest_mesh.GetNodes()->FESpace()->GetOrdering();
+  }
+  else
+  {
+    int npts = 0, offset = 0;
+    for (int i = 0; i < dest_mesh.GetNE(); i++)
+    {
+      npts += V.FESpace()->GetFE(i)->GetNodes().GetNPoints();
```

Comment:
Yeah the latter, we just find the duplicate shared points nodes between elements and they are disregarded because we `SetSubVector` later on. Certainly a future optimization could be to remove these duplicates from the GSLIB search and just maintain the indexing separately.